<launch>
  <arg name="goal_topic_name" default="/planning/mission_planning/goal"/>
  <arg name="checkpoint_topic_name" default="/planning/mission_planning/checkpoint"/>
  <arg name="rout_topic_name" default="/planning/mission_planning/route"/>
  <arg name="map_topic_name" default="/map/vector_map"/>
  <arg name="visualization_topic_name" default="/planning/mission_planning/route_marker"/>
  <arg name="lane_change_param" default="$(find-pkg-share route_publisher_pm2)/param/lane_change.param.yaml"/>
  <arg name="vehicle_param_file" default="$(find-pkg-share sample_vehicle_description)/config/vehicle_info.param.yaml"/>

  <node pkg="route_publisher_pm2" exec="route_publisher_pm2" name="route_publisher_pm2" output="screen">
    <param from="$(var lane_change_param)"/>
    <param from="$(var vehicle_param_file)"/>
    <param name="map_frame" value="map"/>
    <param name="base_link_frame" value="base_link"/>
    <remap from="input/vector_map" to="$(var map_topic_name)"/>
    <remap from="input/goal_pose" to="$(var goal_topic_name)"/>
    <remap from="input/checkpoint" to="$(var checkpoint_topic_name)"/>
    <remap from="output/route" to="$(var rout_topic_name)"/>
    <remap from="debug/route_marker" to="$(var visualization_topic_name)"/>
  </node>
</launch>
